core/src/Kokkos_MathematicalSpecialFunctions.hpp

```diff
@@ -503,7 +503,7 @@ KOKKOS_INLINE_FUNCTION CmplxType cyl_bessel_j0(const CmplxType& z,
       for (int k = bw_start; k >= 0; k--) {  // Backward recurrence (default:
                                              // 70)
         cf                    = 2.0 * (k + 1.0) / z * cf1 - cf2;
-        RealType tmp_exponent = static_cast<RealType>(k / 2);
+        RealType tmp_exponent = static_cast<RealType>(k) / 2;
```

Comment:
Why is this change correct? On first sight it seems integer division is intended, in particular since we use more integer devision below.

Same in the rest of the file.

Reply:
That's a good question.
I guess I originally wondered if that cast was superfluous or if it was a mistake and I changed out of curiosity to see if tests would fail.  I forgot about it since and I should have highlighted it properly.
Thanks for bringing that up.

Reply:
The algorithms in https://people.sc.fsu.edu/~jburkardt/f_src/special_functions/special_functions.f90 also used integer division so I restored that behavior.